src/main/com/fulcrologic/fulcro/algorithms/normalized_state_helpers.cljc

```diff
@@ -0,0 +1,539 @@
+(ns com.fulcrologic.fulcro.algorithms.normalized-state-helpers
+  "Functions that can be used against a normalized Fulcro state database."
+  #?(:cljs (:require-macros com.fulcrologic.fulcro.algorithms.normalized-state-helpers))
+  (:refer-clojure :exclude [get-in])
+  (:require
+    [com.fulcrologic.fulcro.components :as comp :refer [defsc]]
+    #?(:clj  [com.fulcrologic.fulcro.dom-server :as dom]
+       :cljs [com.fulcrologic.fulcro.dom :as dom])
+    [edn-query-language.core :as eql]
+    [clojure.spec.alpha :as s]
+    [ghostwheel.core :refer [>defn =>]]
+    [com.fulcrologic.fulcro.algorithms.data-targeting :as targeting]
+    [com.fulcrologic.fulcro.algorithms.denormalize :as fdn]
+    [com.fulcrologic.fulcro.algorithms.merge :as merge]
+    [com.fulcrologic.fulcro.components :as comp]))
+
+
+
+;;;;;;;;;;;;;;;;;;
+;; TODO
+
+;; - [ ] Test this namespace works in CLJC
+;; - [ ]  Learn how to run the ghostwheel test in the browser
+;; - [ ] Try out generative testing
+
+
+;; REMOVE this
+(def normalized-state {:car/id      {1 {:car/id 1, :car/model "T-150"},
+                                     2 {:car/id 2, :car/model "Ferrari"},
```

Comment:
These (which I'm guessing you're using from the REPL) are commonly placed in a lisp-structured comment at the bottom of the file, and can easily be used that way:

```
(comment
   (def a ...)
   ...)
```

Such comments have to be legal syntax, but then support full editor and REPL interfacing.

Reply:
Yup however, for some reason Tony, these seem not stay in scope if I don't keep them at the top. This was kind of a lazy solution 😅 

No worries about this though, I'll clear up as per the project's formatting once I'm sure of the solutions.

Reply:
As comments they are never in scope.  You execute them in the order you need them in the REPL for playing around.